services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -243,16 +176,113 @@ def _reraise_if_not_unique_uuid_error(err: UniqueViolation):
+                                    ProjectNodeCreate(
+                                        node_id=NodeID(node_id), required_resources={}
+                                    ),
+                                )
+                                for node_id in selected_values["workbench"]
+                            ]
+                            await project_nodes_repo.add(conn, nodes=nodes)
+        return selected_values
+
+    async def insert_project(
+        self,
+        project: dict[str, Any],
+        user_id: int | None,
+        *,
+        product_name: str,
+        force_project_uuid: bool = False,
+        force_as_template: bool = False,
+        hidden: bool = False,
+        project_nodes: dict[NodeID, ProjectNodeCreate] | None,
+    ) -> dict[str, Any]:
+        """Inserts a new project in the database
+
+        - A valid uuid is automaticaly assigned to the project except if force_project_uuid=False. In the latter case,
+        - passing project_nodes=None will auto-generate default ProjectNodeCreate, default resources will then be used.
+        invalid uuid will raise an exception.
 
+        :raises ProjectInvalidRightsError: assigning project to an unregistered user
+        :raises ValidationError
+        :return: inserted project
+        """
+
+        # NOTE: tags are removed in convert_to_db_names so we keep it
+        project_tag_ids = parse_obj_as(list[int], project.get("tags", []).copy())
+        insert_values = convert_to_db_names(project)
+        insert_values.update(
+            {
+                "type": ProjectType.TEMPLATE.value
+                if (force_as_template or user_id is None)
+                else ProjectType.STANDARD.value,
+                "prj_owner": user_id if user_id else None,
+                "hidden": hidden,
+                # NOTE: this is very bad and leads to very weird conversions.
+                # needs to be refactored!!! use arrow (https://github.com/ITISFoundation/osparc-simcore/issues/3797)
+                "creation_date": now_str(),
+                "last_change_date": now_str(),
+            }
+        )
+
+        # validate access_rights. are the gids valid? also ensure prj_owner is in there
+        if user_id:
+            async with self.engine.acquire() as conn:
+                primary_gid = await self._get_user_primary_group_gid(
+                    conn, user_id=user_id
+                )
+            insert_values.setdefault("access_rights", {})
+            insert_values["access_rights"].update(
+                create_project_access_rights(primary_gid, ProjectAccessRights.OWNER)
+            )
+
+        # ensure we have the minimal amount of data here
+        # All non-default in projects table
+        insert_values.setdefault("name", "New Study")
+        insert_values.setdefault("workbench", {})
+
+        # must be valid uuid
+        try:
+            uuidlib.UUID(str(insert_values.get("uuid")))
```

Comment:
just noticed this, what is the reason for `import uuid as uuidlib`above ?
can't we remove it? I find it very strange, thought this uuidlib was some special library.

Reply:
well I am not the author of this. let me check into that.

Reply:
ok refactored to just uuid1()